source/containerimage/pull.go

```diff
@@ -203,14 +205,31 @@ func (p *puller) CacheKey(ctx context.Context, g session.Group, index int) (cach
 				progressController.Name = p.vtx.Name()
 			}
 
-			descHandler := &cache.DescHandler{
-				Provider: p.manifest.Remote.Provider,
-				Progress: progressController,
+			var layers string
+			for _, desc := range p.manifest.Remote.Descriptors {
+				layers += fmt.Sprintf("%s,", desc.Digest.String())
+			}
+			if len(layers) >= 1 {
```

Comment:
use `strings.TrimSuffix` to trim the extra `,`.

Or consider using `strings.Join`